xcode/Subconscious/Shared/Components/Common/Byline/PetnameView.swift

```diff
@@ -7,54 +7,111 @@
 
 import SwiftUI
 
+public enum NameVariant {
+    case known(Slashlink, Petname.Name)
+    case named(Slashlink, Petname.Name)
+    case unknown(Slashlink)
+}
+
+struct PeerView: View {
+    var peer: Peer
+
+    var body: some View {
+        HStack(spacing: AppTheme.unit) {
+            ZStack {
+                RoundedRectangle(cornerRadius: AppTheme.cornerRadius/2)
+                    .foregroundColor(.secondaryBackground)
+                Text("AKA")
+                    .foregroundColor(.secondary)
+                    .font(.system(size: 10))
+            }
+            .frame(width: 28, height: 15)
+
+            Text(peer.markup)
+                .foregroundColor(.secondary)
+                .fontWeight(.regular)
+                .font(.caption)
+        }
+    }
+}
+
 /// Byline style for displaying a petname
 struct PetnameView: View {
-    var address: Slashlink
-    var name: Petname?
+    var name: NameVariant
+    var annotation: String? = nil
+    
+    public func annotated(with annotation: String) -> Self {
```

Comment:
For localization reasons, we want to express this as a string interpolation via `LocalizedStringKey`.

Given it's an interpolation, I don't think we can to express this as a generalized `annotated` method. Rather, we probably want to do the interpolation close to, or at the call site, probably in `PetnameView` body.

```swift
Text("Maybe: \(name)")
```

https://developer.apple.com/documentation/swiftui/localizedstringkey/init(stringinterpolation:)